drivers/display/ssd16xx.c

```diff
@@ -588,17 +634,30 @@ static int ssd16xx_set_pixel_format(const struct device *dev,
 static int ssd16xx_set_orientation(const struct device *dev,
 				   const enum display_orientation orientation)
 {
+	const struct ssd16xx_config *config = dev->config;
 	struct ssd16xx_data *data = dev->data;
 	int err;
 
-	if (orientation == DISPLAY_ORIENTATION_NORMAL) {
-		data->scan_mode = SSD16XX_DATA_ENTRY_XDYIY;
-	} else if (orientation == DISPLAY_ORIENTATION_ROTATED_90) {
-		data->scan_mode = SSD16XX_DATA_ENTRY_XDYDX;
-	} else if (orientation == DISPLAY_ORIENTATION_ROTATED_180) {
-		data->scan_mode = SSD16XX_DATA_ENTRY_XIYDY;
-	} else if (orientation == DISPLAY_ORIENTATION_ROTATED_270) {
-		data->scan_mode = SSD16XX_DATA_ENTRY_XIYIX;
+	if (config->scan_y_reverse) {
+		if (orientation == DISPLAY_ORIENTATION_NORMAL) {
+			data->scan_mode = SSD16XX_DATA_ENTRY_XDYDY;
+		} else if (orientation == DISPLAY_ORIENTATION_ROTATED_90) {
+			data->scan_mode = SSD16XX_DATA_ENTRY_XIYDX;
+		} else if (orientation == DISPLAY_ORIENTATION_ROTATED_180) {
+			data->scan_mode = SSD16XX_DATA_ENTRY_XIYIY;
+		} else if (orientation == DISPLAY_ORIENTATION_ROTATED_270) {
+			data->scan_mode = SSD16XX_DATA_ENTRY_XDYIX;
+		}
+	} else {
+		if (orientation == DISPLAY_ORIENTATION_NORMAL) {
+			data->scan_mode = SSD16XX_DATA_ENTRY_XDYIY;
+		} else if (orientation == DISPLAY_ORIENTATION_ROTATED_90) {
+			data->scan_mode = SSD16XX_DATA_ENTRY_XDYDX;
+		} else if (orientation == DISPLAY_ORIENTATION_ROTATED_180) {
+			data->scan_mode = SSD16XX_DATA_ENTRY_XIYDY;
+		} else if (orientation == DISPLAY_ORIENTATION_ROTATED_270) {
+			data->scan_mode = SSD16XX_DATA_ENTRY_XIYIX;
+		}
```

Comment:
More of a suggestion:
```suggestion
		if (orientation == DISPLAY_ORIENTATION_NORMAL) {
			data->scan_mode = config->scan_y_reverse ? SSD16XX_DATA_ENTRY_XDYDY : SSD16XX_DATA_ENTRY_XDYIY;
		} else if (orientation == DISPLAY_ORIENTATION_ROTATED_90) {
			data->scan_mode = config->scan_y_reverse ? SSD16XX_DATA_ENTRY_XIYDX : SSD16XX_DATA_ENTRY_XDYDX;
		} else if (orientation == DISPLAY_ORIENTATION_ROTATED_180) {
			data->scan_mode = config->scan_y_reverse ? SSD16XX_DATA_ENTRY_XIYIY : SSD16XX_DATA_ENTRY_XIYDY;
		} else if (orientation == DISPLAY_ORIENTATION_ROTATED_270) {
			data->scan_mode = config->scan_y_reverse ? SSD16XX_DATA_ENTRY_XDYIX : SSD16XX_DATA_ENTRY_XIYIX;
		}
```